app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -291,4 +291,64 @@ public function validateBeforeOrderCommit($observer) {
         $orderModel = Mage::getModel('boltpay/order');
         $orderModel->validateBeforeOrderCommit($observer);
     }
+
+    /**
+     * Automatically create an invoice after creating the shipment if createInvoiceAfterCreatingShipment setting is set to yes
+     * @param $observer
+     * @return $this
+     */
+    public function createInvoiceAfterCreatingShipment($observer)
+    {
+        $shipment = $observer->getEvent()->getShipment();
+        $order = $shipment->getOrder();
+        if (
+            !$this->boltHelper()->getExtraConfig('createInvoiceAfterCreatingShipment')
+            || !Mage::getModel('boltpay/order')->isBoltOrder($order)
+        ) {
+            return;
+        }
+
+        try {
+            if ($order->canInvoice()) {
+                $invoiceItems = array();
+                $shipmentItems = $shipment->getAllItems();
+                foreach ($shipmentItems as $shipmentItem) {
+                    $qty = $this->getApplicableQtyToInvoice($order, $shipmentItem);
+                    if ($qty > 0) {
+                        $invoiceItems[$shipmentItem->getOrderItemId()] = $qty;
+                    }
+                }
+                if(count($invoiceItems) > 0){
+                    $invoice = Mage::getModel('sales/service_order', $order)->prepareInvoice($invoiceItems);
+                    $invoice->setRequestedCaptureCase(Mage_Sales_Model_Order_Invoice::CAPTURE_ONLINE);
+                    $invoice->register();
+                    $transaction = Mage::getModel('core/resource_transaction')
+                        ->addObject($invoice)
+                        ->addObject($order);
+                    $transaction->save();
+                }
+            }
+        } catch (\Exception $exception) {
+            $this->boltHelper()->logException($exception);
+            $this->boltHelper()->notifyException($exception);
+        }
+    }
+
+    /**
+     * @param $order
+     * @param $shipmentItem
+     * @return bool|float
+     */
+    private function getApplicableQtyToInvoice($order, $shipmentItem)
+    {
+        foreach ($order->getAllVisibleItems() as $orderItem) {
```

Comment:
Maybe looping through all items is not the most efficient way to look up an item from an order if we already have the order item ID.  Still, the performance gain from doing this another way is probably negligible.